### ISSUE TITLE
python3 support

### ISSUE THREAD

**Issue by shukti02** [FCP-INDI/C-PAC#942]
Dear developers, approximately when will the python3 compatible version be available?
Thanks

**Comment by sgiavasis** [comment]
Hello @shukti02,

Currently, the plan is to make the transition to python3 over several releases. The timeline has not yet been finalized, but we're looking at a few months.

Best,
Steve

**Comment by shukti02** [comment]
Okay, thank you. CPAC installation on Ubuntu has a lot of library
compatibility issues, even if we follow the documentation on the internet.

The current one I'm facing is with configparser, because the function ga.py
calls a function read_dict of the configparser, which is only available in
the python3 version and not the python2 one. Any workarounds for this?

Thanks in advance.

Best Regards
Shukti

**Comment by anibalsolon** [comment]
Hi @shukti02 

For the configparser, you can install a backport for Python 2: `pip install configparser`
We will add it as a project dependency, thanks for reporting it.

However, we strongly recommend using containers (Docker or Singularity), if that is a possibility. They deal with differences in software versions and potential incompatibilities/misbehaviors, increasing the reproducibility of your study.

Please let us know about other incompatibilities you detected in the installation process.

**Comment by shukti02** [comment]
Hi Anibal,

Thanks for the reply. I am using miniconda for the cpac installation.

Both the pip install and conda install don't seem get the right version of
configparser.

I tried another workaround, by copying the configparser package from
backports into the packages directory of my environment, but this doesn't
work as well as it uses a python3 syntax and throws other errors then.

I am unable to solve this issue, can you please suggest other workarounds
to it?

Thanks and Regards
Shukti

**Comment by anibalsolon** [comment]
Hi @shukti02 

Actually, this workaround was supposed to solve your issue.

Could you please paste here the full output of this command?

```
python -c "import sys, configparser, CPAC; print(sys.version_info); print(CPAC.__version__); print(configparser); print(configparser.ConfigParser.read_dict)"
```

**Comment by shukti02** [comment]
Hi Anibal,

With the backports workaround which i previously wrote, the output of the
above command is:

/home/shukti/miniconda2/envs/cpac/lib/python2.7/site-packages/CPAC/connectome/pipeline.py:13:
DeprecationWarning: Python2 support is deprecated and will be removed in a
future release. Consider switching to Python3.
  from nilearn.connectome import ConnectivityMeasure
sys.version_info(major=2, minor=7, micro=16, releaselevel='final', serial=0)
1.4.1
<module 'configparser' from
'/home/shukti/miniconda2/envs/cpac/lib/python2.7/site-packages/configparser/__init__.pyc'>
<unbound method ConfigParser.read_dict>

Thanks and Regards
Shukti

**Comment by shukti02** [comment]
Hi,

Sorry, i just realized that this workaround solved the problem of
configparser, and the error which i am getting now is something different.
The Nipype logging module doesn't have an attribute called info while the
default logging module does. cpac_pipeline.py however calls the nipype
module instead of the default module.

Best Regards
Shukti

**Comment by anibalsolon** [comment]
I would assume that is about the nipype version.

Could you please run `python -c "import nipype; print(nipype.__version__)"` ?

**Comment by shukti02** [comment]
My current Nipype version is 1.0.3. I had to upgrade it from 0.13.1 (the
version suggested by the documentation) as CPAC imports fsl.aroma which is
only available in nipype versions 1+, and not available in 0.13.1.

However I thought about this too, that the logging error could be due to an
incompatible nipype version so i tried downgrading it to 0.13.1 again.
However, this logging error persists even in Nipype 0.13.1, therefore i
switched back to 1.0.3.

Best Regards
Shukti

**Comment by shukti02** [comment]
Hi Anibal,

Thank you  very much for the prompt support and help.

I checked all package dependencies in the info.py file of the CPAC package
and re-installed package versions accordingly, now CPAC is running
successfully on my PC.

I would recommend the developers of CPAC to update the  online
documentation, or to mention that dependencies can be checked in this file
('info.py') so that others installing it in future do not face the same
problems as I did. The current dependency versions are very different from
the ones mentioned in the documentation.

Thank you once again.

Best Regards
Shukti
